frontend/src/metabase/visualizer/utils/merge-data.ts

```diff
@@ -50,16 +68,18 @@ export function mergeVisualizerData({
         }
         const values = referencedColumnValuesMap[valueSource.name];
         if (!values) {
-          return [];
+          return undefined;
         }
         return values;
       })
       .flat(),
   );
 
-  return {
-    cols: columns,
-    rows: _.zip(...unzippedRows),
-    results_metadata: { columns },
-  };
+  return unzippedRows.some((v) => !v || v.some((_v) => _v === undefined))
+    ? undefined
+    : {
+        cols: columns,
+        rows: _.zip(...unzippedRows),
+        results_metadata: { columns },
+      };
```

Comment:
nit: it's somewhat hard to parse what we're doing inside `.some(...)`, maybe we could make it a bit more clear if we extract the arrow function and give it a name, maybe something like:

```ts
const isEmptyRow = val => !val || val.some(v => v === undefined))

// ...

return unzippedRows.some(isEmptyRow) ? ... : ...
```